python/gsw/main.py

```diff
@@ -0,0 +1,370 @@
+        radio_keys_config = json.load(radio_keys_config_file)
+    with open(os.path.abspath(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../usb_console/configs/server_keys.json')))as server_keys_config_file:
+        server_keys_config = json.load(server_keys_config_file)
+except json.JSONDecodeError:
+    print("Could not load config files. Exiting.")
+    raise SystemExit
+except KeyError:
+    print("Malformed config file. Exiting.")
+    raise SystemExit
+
+# Create a read_iridium object
+readIr=read_iridium(radio_keys_config, server_keys_config)
+# Start checking emails and posting reports
+readIr.connect()
+
+# Set up the SwaggerUI API
+swagger_config={
+    "headers":[],
+    "specs":[
+        {
+            "endpoint":"apispec_1",
+            "route":"/apispec_1.json",
+            "rule_filter":lambda rule:True,
+            "model_filter":lambda tag:True
+        }
+    ],
+    "static_url_path": "/flassger_static",
+    "swagger_ui":True,
+    "specs_route":"/swagger/"
+}
+swagger=Swagger(app, config=swagger_config)
+
+# Endpoint for indexing data in elasticsearch
+# Mostly for testing purposes. We don't use this to actually index data in elasticsearch
+@app.route("/telemetry", methods=["POST"])
+@swag_from("endpoint_configs/telemetry_config.yml")
+def index_sf_report():
+    #connect to elasticsearch
+    es=readIr.es
+
+    sf_report=request.get_json()
+    #index statefield report in elasticsearch
+    sf_res = es.index(index='statefield_report', doc_type='report', body=sf_report)
+    res={"Report Status": sf_res['result']}
+    return res
+
+# Endpoint for getting data from the statefield reports index in elasticsearch
+@app.route("/search-statefields", methods=["GET"])
+@swag_from("endpoint_configs/search_statefields_config.yml")
+def get_statefield():
+    # Connect to elasticsearch
+    es=readIr.es
+
+    imei = int(request.args.get('imei'))
+    statefield = str(request.args.get('statefield'))
+
+    # Get the most recent document in the statefield index which has a given statefield in it
+    search_object={
+        'query': {
+            'exists': {
+                'field': statefield
+            }
+        },
+        "sort": [
+            {
+                "time": {
+                    "order": "desc"
+                }
+            }
+        ],
+        "size": 1
+    }
+    
+    # Get the value of that statefield from the document
+    if es.indices.exists(index='statefield_report_'+str(imei)):
+        res = es.search(index='statefield_report_'+str(imei), body=json.dumps(search_object))
+        if len(res["hits"]["hits"])!=0:
+            most_recent_field=res["hits"]["hits"][0]["_source"][statefield]
+            return most_recent_field
+        else:
+            return "Unable to find field in index: "+"statefield_report_"+str(imei)
+    else:
+        return "Unable to find index: "+"statefield_report_"+str(imei)
```

Comment:
You're using very similar code below. Consider breaking this out into a function

Reply:
If I broke all this out into a separate function, I would basically be creating a single endpoint to handle searching both iridium and statefield reports in elasticsearch. If I combined these two endpoints into a single endpoint, I would have to add another query parameter specifying whether to search the iridium index or the statefield index. Personally I think it is simpler to keep two separate endpoints for searching statefield reports and iridium reports and avoid adding that extra parameter. Do you think it would be better to combine them into a single endpoint?

Reply:
Upon further reflection I realized i could just combine these two into a single endpoint